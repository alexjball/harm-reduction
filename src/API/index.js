export { db, auth } from './Firebase';
export {
  addParticipant,
  addIncident,
  addEngagement,
  getParticipantById,
  getIncidentsByParticipantId,
} from './FirebaseModel';
