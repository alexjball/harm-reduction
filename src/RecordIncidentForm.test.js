import { render, act, getByLabelText } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import React from 'react';
import { db, addIncident } from './API';
import RecordIncidentForm from './RecordIncidentForm';
import { stateOfChange } from './Options';

jest.mock('./API');

test('Renders Incident Form', () => {
  const incidentForm = render(<RecordIncidentForm />);
  expect(incidentForm).toMatchSnapshot();
});

test.each([
  ['First Name', 'Name', 'Name'],
  ['Last Name', 'Name', 'Name'],
  ['Location (City/Town)', 'Location', 'Location'],
])('%s takes input', (a, b, expected) => {
  const { getByLabelText } = render(<RecordIncidentForm />);
  userEvent.type(getByLabelText(a), b);
  expect(getByLabelText(a)).toHaveValue(expected);
});

test('Date field accepts date', async () => {
  const { getByLabelText } = render(<RecordIncidentForm />);
  const dateInput = getByLabelText('Date of Request');
  const date = '2020-12-12';

  await userEvent.type(dateInput, date);

  expect(dateInput).toHaveValue(date);
});

test('Can check narcan box', async () => {
  const { getByLabelText } = render(<RecordIncidentForm />);
  await userEvent.click(getByLabelText('Recieved Narcan'));
  expect(getByLabelText('Recieved Narcan')).toBeChecked();
});
