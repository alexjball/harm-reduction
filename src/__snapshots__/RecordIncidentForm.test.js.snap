// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Incident Form 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div
          style="width: 50%; margin-left: 5%; margin-top: 5%; margin-bottom: 5%;"
        >
          <div
            style="display: flex; align-items: baseline; justify-content: flex-start; margin-bottom: 2em;"
          >
            <h2
              style="flex: 1;"
            >
              Add a new incident
            </h2>
            <p
              style="flex: 1; font-weight: 400; font-size: 1em;"
            >
              <em
                data-testid="submissionstatus"
              >
                Waiting for input
              </em>
            </p>
          </div>
          <form
            class=""
            data-testid="submitform"
            style="margin-bottom: 20px;"
          >
            <div
              class="row"
            >
              <div
                class="form-group col"
              >
                <label
                  class="form-label"
                  for="firstName"
                >
                  First Name
                </label>
                <input
                  class="form-control"
                  id="firstName"
                  value=""
                />
              </div>
              <div
                class="form-group col"
              >
                <label
                  class="form-label"
                  for="lastName"
                >
                  Last Name
                </label>
                <input
                  class="form-control"
                  id="lastName"
                  type="text"
                  value=""
                />
              </div>
            </div>
            <div
              class="form-group"
            >
              <label
                class="form-label"
                for="participantId"
              >
                Participant ID
              </label>
              <input
                class="form-control"
                id="participantId"
                required=""
                type="text"
                value=""
              />
            </div>
            <div
              class="row"
            >
              <div
                class="form-group col"
              >
                <label
                  class="form-label"
                  for="location"
                >
                  Location (City/Town)
                </label>
                <input
                  class="form-control"
                  id="location"
                  type="text"
                  value=""
                />
              </div>
            </div>
            <div
              class="row"
              style="align-items: center;"
            >
              <div
                class="form-group col"
              >
                <label
                  class="form-label"
                  for="dateOfRequest"
                >
                  Date of Request
                </label>
                <input
                  class="form-control"
                  id="dateOfRequest"
                  type="date"
                  value=""
                />
              </div>
              <div
                class="col"
              >
                <div
                  class="form-check"
                  style="vertical-align: middle;"
                >
                  <input
                    class="form-check-input"
                    id="recievedNarcan"
                    type="checkbox"
                  />
                  <label
                    class="form-check-label"
                    for="recievedNarcan"
                    title=""
                  >
                    Recieved Narcan
                  </label>
                </div>
              </div>
            </div>
            <div
              class="form-group"
            >
              <label
                class="form-label"
                for="notes"
              >
                Notes
              </label>
              <textarea
                class="form-control"
                id="notes"
                rows="7"
              />
            </div>
            <button
              class="btn btn-primary"
              type="submit"
            >
              Record Incident
            </button>
             
            <p>
              STATUS: 
              Waiting for input
            </p>
          </form>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        style="width: 50%; margin-left: 5%; margin-top: 5%; margin-bottom: 5%;"
      >
        <div
          style="display: flex; align-items: baseline; justify-content: flex-start; margin-bottom: 2em;"
        >
          <h2
            style="flex: 1;"
          >
            Add a new incident
          </h2>
          <p
            style="flex: 1; font-weight: 400; font-size: 1em;"
          >
            <em
              data-testid="submissionstatus"
            >
              Waiting for input
            </em>
          </p>
        </div>
        <form
          class=""
          data-testid="submitform"
          style="margin-bottom: 20px;"
        >
          <div
            class="row"
          >
            <div
              class="form-group col"
            >
              <label
                class="form-label"
                for="firstName"
              >
                First Name
              </label>
              <input
                class="form-control"
                id="firstName"
                value=""
              />
            </div>
            <div
              class="form-group col"
            >
              <label
                class="form-label"
                for="lastName"
              >
                Last Name
              </label>
              <input
                class="form-control"
                id="lastName"
                type="text"
                value=""
              />
            </div>
          </div>
          <div
            class="form-group"
          >
            <label
              class="form-label"
              for="participantId"
            >
              Participant ID
            </label>
            <input
              class="form-control"
              id="participantId"
              required=""
              type="text"
              value=""
            />
          </div>
          <div
            class="row"
          >
            <div
              class="form-group col"
            >
              <label
                class="form-label"
                for="location"
              >
                Location (City/Town)
              </label>
              <input
                class="form-control"
                id="location"
                type="text"
                value=""
              />
            </div>
          </div>
          <div
            class="row"
            style="align-items: center;"
          >
            <div
              class="form-group col"
            >
              <label
                class="form-label"
                for="dateOfRequest"
              >
                Date of Request
              </label>
              <input
                class="form-control"
                id="dateOfRequest"
                type="date"
                value=""
              />
            </div>
            <div
              class="col"
            >
              <div
                class="form-check"
                style="vertical-align: middle;"
              >
                <input
                  class="form-check-input"
                  id="recievedNarcan"
                  type="checkbox"
                />
                <label
                  class="form-check-label"
                  for="recievedNarcan"
                  title=""
                >
                  Recieved Narcan
                </label>
              </div>
            </div>
          </div>
          <div
            class="form-group"
          >
            <label
              class="form-label"
              for="notes"
            >
              Notes
            </label>
            <textarea
              class="form-control"
              id="notes"
              rows="7"
            />
          </div>
          <button
            class="btn btn-primary"
            type="submit"
          >
            Record Incident
          </button>
           
          <p>
            STATUS: 
            Waiting for input
          </p>
        </form>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
